monkey/infection_monkey/exploit/powershell_utils/powershell_client.py

```diff
@@ -89,27 +83,22 @@ def __init__(self, ip_addr, credentials: Credentials, auth_options: AuthOptions)
             connection_timeout=CONNECTION_TIMEOUT,
         )
 
+        # Attempt to execute dir command to know if authentication was successful. This will raise
+        # an exception if authentication was not successful.
+        self.execute_cmd("dir")
+        logger.debug("Successfully authenticated to remote PowerShell service")
```

Comment:
Why is this constructor executing commands? Constructor's have one job and it is to construct an object.

Reply:
I added `connect()` to `IPowerShellClient` to resolve this.